diagnostics/albedofb/albedofb_calcs.py

```diff
@@ -0,0 +1,207 @@
+#          The Effect of Atmospheric Transmissivity on Model and Observational Estimates 
+#          of the Sea Ice Albedo Feedback, J. Climate, https://doi.org/10.1175/JCLI-D-19-0674.1
+#
+
+import os
+
+# Commands to load third-party libraries. Any code you don't include that's 
+# not part of your language's standard library should be listed in the 
+# settings.jsonc file.
+import xarray as xr                # python library we use to read netcdf files
+import numpy as np
+import pandas as pd
+
+
+def xr_reshape(A, dim, newdims, coords):
+    """ Reshape DataArray A to convert its dimension dim into sub-dimensions given by
+    newdims and the corresponding coords.
+    Example: Ar = xr_reshape(A, 'time', ['year', 'month'], [(2017, 2018), np.arange(12)]) """
+
+    # Create a pandas MultiIndex from these labels
+    ind = pd.MultiIndex.from_product(coords, names=newdims)
+
+    # Replace the time index in the DataArray by this new index,
+    A1 = A.copy()
+
+    A1.coords[dim] = ind
+
+    # Convert multiindex to individual dims using DataArray.unstack().
+    # This changes dimension order! The new dimensions are at the end.
+    A1 = A1.unstack(dim)
+
+    # Permute to restore dimensions
+    i = A.dims.index(dim)
+    dims = list(A1.dims)
+
+    for d in newdims[::-1]:
+        dims.insert(i, d)
+
+    for d in newdims:
+        _ = dims.pop(-1)
+
+    return A1.transpose(*dims)
+
+def climatology(field=None):
+    """Compute mean, std, trend, std of detrended, residuals
+
+    Parameters
+    ----------
+    field : xarray.DataArray, dims must be time, space (space can be multidim)
+    
+    Returns
+    -------
+    themean, thestd, trend, detrendedstd: xarray.DataArray, dims of month, space
+    residuals: xarray.DataArray, dims of year, month, space
+    """
+    
+    NY=int(field['time'].sizes['time']/12)
+
+    field=xr_reshape(field,'time',['year','month'],[np.arange(NY),np.arange(12)])
+    themean = field.mean(dim='year')
+
+    return themean
+
+def kernelalbedo(dt, ds, ut, us):
+    # solve for single layer reflections under Taylor '07 assumptions
+
+    alb = us/ds #surface albedo
+    absorb=(dt-ut-ds+us)/dt #atmospheric absorption (assumed first pass only)
+    R = (1-absorb-ds/dt)/(1-absorb-ds*alb/dt) #anayltitcally solution for single layer atmospheric reflection
+
+    kernel = (1./100.)*(dt*((1.-R)*(1.-R))*(1.-absorb)*(1.-alb*R+R))/((1.-R*alb)*(1.-R*alb)) #derivative of upwelling solar at TOA with respect to surface albedo --  in units of W m^-2 per .01 change in surface albedo
+    kernel.name = 'kernel'
+    
+    albedo = us/ds
+    albedo.name = 'albedo'
+
+    return kernel, albedo
+
+def globaltimemean(tas, fx):
+    t=tas.mean(dim='time')
+    glob=t*fx
+    glob=glob.sum()/fx.sum()
+    glob.name='Tglob'
+    
+    return glob
+
+def globalmean(tas, fx):
+    glob=tas*fx
+    glob=glob.sum(dim=["lat", "lon"])/fx.sum(dim=["lat", "lon"])
+    glob.name='Tglob'
+    
+    return glob
```

Comment:
These are fine as-is, but could be replaced by xarray's `.weighted()` method in the future.

Reply:
Noted